tips/TIP-0020/tip-0020.md

```diff
@@ -0,0 +1,497 @@
+| Alias             | 2          | [draft TIP-18](https://github.com/lzpap/protocol-rfcs/blob/master/tips/TIP-0018/tip-0018.md#alias-locking--unlocking) |
+| NFT               | 3          | [draft TIP-18](https://github.com/lzpap/protocol-rfcs/blob/master/tips/TIP-0018/tip-0018.md#nft-locking--unlocking)   |
+
+#### Signature Unlock Block
+
+The <i>Signature Unlock Block</i> defines an <i>Unlock Block</i> which holds a signature signing the BLAKE2b-256 hash of the <i>Transaction Essence</i> (including the optional payload). It is serialized as follows:
+
+<table>
+  <tr>
+    <th>Name</th>
+    <th>Type</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>Unlock Type</td>
+    <td>uint8</td>
+    <td>
+      Set to <strong>value 0</strong> to denote a <i>Signature Unlock Block</i>.
+    </td>
+  </tr>
+  <tr>
+    <td valign="top">Signature <code>oneOf</code></td>
+    <td colspan="2">
+      <details>
+        <summary>Ed25519 Signature</summary>
+        <table>
+          <tr>
+            <th>Name</th>
+            <th>Type</th>
+            <th>Description</th>
+          </tr>
+          <tr>
+            <td>Signature Type</td>
+            <td>uint8</td>
+            <td>
+              Set to <strong>value 0</strong> to denote an <i>Ed25519 Signature</i>.
+            </td>
+          </tr>
+          <tr>
+            <td>Public key</td>
+            <td>ByteArray[32]</td>
+            <td>The Ed25519 public key of the signature.</td>
+          </tr>
+          <tr>
+            <td>Signature</td>
+            <td>ByteArray[64]</td>
+            <td>The Ed25519 signature signing the Blake2b-256 hash of the serialized <i>Transaction Essence</i>.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+</table>
+
+##### Unlock syntactic validation
+
+* `Signature` must contain an _Ed25519 Signature_.
+* The _Signature Unlock Block_ must be unique, i.e. there must not be any other _Signature Unlock Blocks_ in the `Unlock Blocks` field of the transaction payload with the same signature.
+
+#### Reference Unlock Block
+
+The <i>Reference Unlock Block</i> defines an <i>Unlock Block</i> which references a previous <i>Unlock Block</i> (which must not be another <i>Reference Unlock Block</i>). It **must** be used if multiple inputs can be unlocked via the same <i>Unlock Block</i>. It is serialized as follows:
+
+<table>
+  <tr>
+    <th>Name</th>
+    <th>Type</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>Unlock Type</td>
+    <td>uint8</td>
+    <td>
+      Set to <strong>value 1</strong> to denote a <i>Reference Unlock Block</i>.
+    </td>
+  </tr>
+  <tr>
+    <td>Reference</td>
+    <td>uint16</td>
+    <td>Represents the index of a previous unlock block.</td>
+  </tr>
+</table>
+
+##### Unlock syntactic validation
+
+* The _Reference Unlock Block_ at index i must have `Reference` < i and the unlock block at index `Reference` must be a _Signature Unlock Block_.
```

Comment:
Only `Reference Unlock Block` is shown here, Alias and Nft Unlock Blocks are missing, of course they're very similar, but they are not mentioned there so it's not clear if the syntactic validation is also required for them/the same

Reply:
Couple lines above, the Unlock Blocks section starts with the following sentence:

> The following table lists all the output types that are currently supported as well as links to the corresponding specification. The _Signature Unlock Block_ as well as the _Reference Unlock Block_ is specified as part of this TIP.

| Unlock Block Name | Type Value | TIP                                                                                                                   |
| ----------------- | ---------- | --------------------------------------------------------------------------------------------------------------------- |
| Signature         | 0          | [TIP-20](#signature-unlock-block)                                                                                     |
| Reference         | 1          | [TIP-20](#reference-unlock-block)                                                                                     |
| Alias             | 2          | [draft TIP-18](https://github.com/lzpap/protocol-rfcs/blob/master/tips/TIP-0018/tip-0018.md#alias-locking--unlocking) |
| NFT               | 3          | [draft TIP-18](https://github.com/lzpap/protocol-rfcs/blob/master/tips/TIP-0018/tip-0018.md#nft-locking--unlocking)   |

Ofc, there is an error in the sentence, as it should read: _...lists all the **unlock blocks** that are currently supported..._
Anyway, this sentence clearly says that only these two unlock blocks are discussed here, the others are linked. Isn't that sufficient?

It is also mentioned at the sytectic validation rules that:
>  * For each unlock block the following must be true:
>    * Each `Unlock Block Type` must match one of the values described under [Unlock Blocks](#unlock-blocks).
>    * The unlock block itself must pass syntactic validation.

Reply:
Well yes, I was only looking for `Alias Unlock Block Syntactic Validation` and didn't found it, because it's only linked, so my bad :carlos: